File: lib/src/sdk_api/models/video_presentation.dart
import 'participant.dart';

/// The VideoPresentation class gathers information about a video presentation.
class VideoPresentation {
  /// The participant who started the presentation.
  Participant owner;

  /// The current video presentation timestamp used for seeking and pausing the video.
  num timestamp;

  /// The URL of the presented video file.
  String url;

  VideoPresentation(this.owner, this.timestamp, this.url);
}

File: test_app/integration_tests/android_mocks/src/main/java/com/voxeet/sdk/services/presentation/video/VideoPresentation.java
package com.voxeet.sdk.services.presentation.video;

import androidx.annotation.NonNull;

import com.voxeet.sdk.services.presentation.PresentationState;

public class VideoPresentation {

    /**
     * Represent the unique key for the presentation. This must be treated as a Primary key to differentiate presentations.
     */
    public final String key;

    /**
     * The URL informing about the location of the presented video.
     */
    public String url;

    /**
     * The current state of the video presentation.
     */
    public PresentationState state;

    /**
     * The latest timestamp informing about the current video position.
     */
    public long lastSeekTimestamp;

    private VideoPresentation() {
        key = "";
        url = "";
    }

    public VideoPresentation(@NonNull String key, @NonNull String url) {
        this.key = key;
        this.url = url;
    }

    public VideoPresentation clone() {
        VideoPresentation information = new VideoPresentation(key, url);
        information.state = state;
        return information;
    }
}
